### DISCUSSION TITLE
collect all about WOLF HEIZTECHNIK hardware...

### DISCUSSION THREAD

**Discussion by ulda** [john30/ebusd#779] [General]
Hi everybody

I started this discussion to collect everything we have about [WOLF Heiztechnik ](https://www.wolf.eu/de-de) hardware.
and this nearly nothing at the moment (2022-12-29). At least I did not find much googling.

The WOLF systems sometimes also announce manufacturer KROMSCHROEDER. Hold these as synonome.

I did not find any documentation of  manufacutrers side of the bus - they keep every docu closed behind ther support portal.

But in the ebus spec there are some defaults documented, and over on [ebus-wiki.org](https://www.ebus-wiki.org/doku.php) they collected the spec documents.
Some of the 05xx and 07xx and 08xx messages can be decoded using [this documentation](https://www.ebus-wiki.org/doku.php/ebus/ebusbefehle).
But until now nobody used this to create some default CSVs to use for unknown systems.

They also startet to collect about the vendor specific command set 50xx but they did not come far and what I found in the 2.x.x part of this repository look better.
It looks to me like someone poked on the user terminal for each setting and then looked at the messages it generates on the bus.
This would be another way to come forward with reverse-engineering the 50xx command set at least to get some data out of it.

The configs of the 2.x.x directory of this repo did not work for me, either. 
Some messages had errors on my CGB-35/KK/BM system. And it was a hazzle to get mqtt autoconfiguration running.
becaus the old names do not fit with the default mqtt-hassio.cfg filtering sheme.

So I startet a fork of the ebusd-configuration repo [here at github](https://github.com/ulda/ebusd-configuration) to put in what I have now: some small set of entities generated from the ebus of my CGB-35 gas heating system.

Now I want you to join to create some working default configuration from this devices.

You can help out
- by editing basic CSVs from the spec docu above
- py poking on your hardware and collecting the messages send when you change parameters
- trying to grab and decode unknown messages on the bus of your hardware
 
We all would be happy if you post your findings here.

**Comment by AndreaRicci94** [comment]
Hi all,

Thank you for your effort. I am in a similar situation because I have at home a Wolf heatpump (CHC/CHA) with Solar module SM-1 and control module BM-2. As you can see in #776 I am encountering difficulties getting the data (also getting devices to properly show up).

I checked out your files in the 22.4.x/de folder and I am currently using them, just restarted ebusd.

Please count me in if you need any further information, as I can debug and give data from the system. Mainly because I want this to work :D

**Comment by ulda** [comment]
My System Layout as Example and a small update on whats going on

I have a CGB-35 with BM integrated and directly controlling the hot water tank 
An additional KM controls the heating circuit pump. no mixer hardware.

I scanned my bus and sorted the devices by id and function.
(the ID of the the systems is not  ASCII , so I tried a special scan message setup for it. I then added the raw output of the 0704 message)
This is what I see:


```
# Brenner CGB-35
# laut spec overview  Master #11 = Brenner 1
address 03: master #11
address 08: slave #11, scanned "MF=Kromschroeder;ID=01 06 33 42 02;SW=09 1b;HW=51 30"
	0a 50 0106334202 091b 5130


# Kaskadenmodul KM
# Konfig 9 nur Heizkreispumpe und Sammlertemperatur
# laut spec overview Master #2 = Regler 1
address 10: master #2
address 15: slave #2, scanned "MF=Kromschroeder;ID=  ;SW=0229;HW=-"
	0a 50 0114000900 0229 ffff

# laut spec overview S50 = Mischer 1
address 50: slave, scanned "MF=Kromschroeder;ID=  ;SW=0229;HW=-"
	0a 50 0114000900 0229 ffff

# laut spec overview S51 = Mischer 2
address 51: slave, scanned "MF=Kromschroeder;ID=  ;SW=0229;HW=-"
	0a 50 0114000900 0229 ffff


# BM
# laut spec overview Master #3 = Regler 2
address 30: master #3
address 35: slave #3, scanned "MF=Kromschroeder;ID=  ;SW=0204;HW=-"
	0a 50 0118000000 0204 ffff

# laut spec overview Master #4 = Regler 3
address 70: master #4
# laut spec overview Slave 75 wäre Raumregler
address 75: slave #4, scanned "MF=Kromschroeder;ID=  ;SW=0204;HW=-"
	0a 50 0118000000 0204 ffff

# master#10 unbesetzt nach alter spec overview / Heizungsregler nach neuer spec
address f1: master #10
gaddress f6: slave #10, scanned "MF=Kromschroeder;ID=  ;SW=0204;HW=-"
	0a 50 0118000000 0204 ffff

# master #16 unbesetzt nach spec overview
address 07: master #16
address 0c: slave #16, scanned "MF=Kromschroeder;ID=  ;SW=0204;HW=-"
	0a 50 0118000000 0204 ffff



# laut spec overview Master #8 = Businterface
address 31: master #8, ebusd
address 36: slave #8, ebusd, scanning

```

I see this  communication on regular basis:
- f1 --> broadcast msg0800 msg0503
- 30 --> broadcast msg5023 msg0700
- 10 --> broadcast msg5023
 
- 03 --> f1 msg0800
 
- 03 --> 10 msg0504
- 10 --> 03 msg0507
 
- 10 --> f1 msg0503
- f1 --> 10  msg0709 msg 0800
 
- 30 --> 50 msg5014
 
- 70 --> 51 msg5014


Most communication uses the speced protocol messages.
Most 5014, 5022 and 5023 communication is unknown - im looking at it.
I can use some of the given 5023 messages on the zz=08 to query temperatures

**Comment by HolyGiraffe** [comment]
Hello! 
I just got my EBusAdapter3 and Setting up everything. 
I have a Wolf CGB-20 with BC MM SM etc etc, and will contribute as soon as my setup is finished.

Currently i cannot get the .csv stuff to work

Is this as it looks at your setup:
/etc/ebusd/de
├── broadcast.csv
├── kromschroeder
│   ├── all.csv
│   └── _templates.csv
├── memory.csv
└── _templates.csv

1 directory, 5 files



Greetings

**Reply by ulda** [reply]
I just pushed my latest working config into my github repro for you to checkout

**Reply by HolyGiraffe** [reply]
> I just pushed my latest working config into my github repro for you to checkout

Ohh my setup was rubbish, its however working now using other .csv's

**Comment by zivillian** [comment]
@ulda you are correct, that there is no publicly available documentation from Wolf regarding their messages on the ebus, but they published a desktop application in the past and are still publishing an android app which contains metadata files with all information needed to map ebus addresses to readable names and even find out which are supported by a concrete setup.

I've "looked closely" at the Wolf Smartset application (and written an [ISM to MQTT Adapter](https://github.com/zivillian/ism7mqtt/)). I've already linked the information in https://github.com/john30/ebusd/issues/460#issuecomment-1028407719 and hoped it would spread further within this community. This tool is not directly talking to the bus, but to an ISM module from Wolf, which basically translates XML to ebus.

In the meantime i also [extracted](https://github.com/zivillian/smartset.net/), [packaged](https://www.nuget.org/packages/smartset.net/) and used the official binaries in a tool to automatically extract all parameters from a given setup without the requirement to run the Wolf app just by connecting to the ISM. So from a user perspective (with the required hardware) it's just:
1. run this binary
2. now run this other binary

and you have read and write access to all parameters

I haven't thought this through, but I'm pretty confident that it should be possible to write some kind of adaptation layer which translates the protocol from the official app to ebusd or some kind of adapter and thus be able to use the app to extract the exact configuration needed for ebusd. So it should be possible to get the same experience of "run this binary to generate a complete config" for users with an ebus adapter/without an ISM.

**Comment by andew311** [comment]
Hi, is this topic still relevant?
I have a Wolf Solar system with an SM2 incl. BM2.
Connected to the ebus is an eBUS Adapter Shield v5.

I actually only want to read out 3 values, pump on or off, temperature in the solar buffer and the modules on the roof. 
But I get a flood of messages on the eBUS. I can't understand which messages are relevant and need to be defined accordingly, or how.
Can someone give me some help?

Thanks

this is how it looks at the moment:

root@vclin:~# cat /var/log/ebusd.log
2024-04-17 00:00:09.151 [main notice] SIGHUP received
2024-04-17 12:01:00.465 [main notice] update check: OK
2024-04-17 12:44:40.457 [bus notice] device status: reset, supports info
2024-04-17 12:44:40.458 [bus notice] device status: reset, supports info
2024-04-17 12:44:40.458 [bus error] device status: transport closed
2024-04-17 12:44:40.458 [bus notice] device invalid
2024-04-17 12:44:45.487 [bus notice] device status: transport opened
2024-04-17 12:44:45.488 [bus notice] re-opened xxx.xxx.xxx.xxx:9999
2024-04-17 12:44:45.488 [bus notice] signal acquired
2024-04-17 12:44:45.490 [bus notice] device status: reset, supports info
2024-04-17 12:44:45.492 [bus notice] device status: extra info: firmware 1.1[431e].1[431e], jumpers 0x0b
2024-04-17 12:44:45.709 [bus notice] new master 71, master count 2
2024-04-17 12:44:45.709 [update notice] received unknown BC cmd: 71fe50171041d90b04d80200800080008000800080
2024-04-17 12:44:49.762 [update notice] received unknown BC cmd: 71fe50230998620901005d010000
2024-04-17 12:44:50.528 [main notice] starting initial broadcast scan
2024-04-17 12:44:50.708 [update notice] sent write broadcast queryexistence QQ=31:
2024-04-17 12:44:56.945 [bus notice] new master 07, master count 3
2024-04-17 12:44:57.681 [update notice] received unknown MS cmd: 0776502203cc0100 / 020000
2024-04-17 12:44:58.020 [update notice] received unknown MS cmd: 0776502203ccc103 / 020100
2024-04-17 12:44:58.372 [update notice] received unknown MS cmd: 0776502203cc4801 / 021500
2024-04-17 12:44:58.727 [update notice] received unknown MS cmd: 0776502203cc6209 / 020100
2024-04-17 12:44:59.085 [update notice] received unknown MS cmd: 0776502203cc2d0a / 020080
2024-04-17 12:44:59.437 [update notice] received unknown MS cmd: 0776502203cc2b0a / 0270fe
2024-04-17 12:44:59.793 [update notice] received unknown MS cmd: 0776502203cc300a / 020080
2024-04-17 12:45:00.130 [update notice] received unknown MS cmd: 0776502203ccae09 / 020000
2024-04-17 12:45:00.483 [update notice] received unknown MS cmd: 0776502203ccf802 / 020000
2024-04-17 12:45:00.836 [update notice] received unknown MS cmd: 0776502203cc2f0a / 020080
2024-04-17 12:45:00.939 [bus notice] scan 0c: ;Wolf; ';0204;8080
2024-04-17 12:45:00.940 [update notice] store 0c ident: done
2024-04-17 12:45:00.940 [update notice] sent scan-read scan.0c  QQ=31: Wolf; ';0204;8080
2024-04-17 12:45:00.940 [bus notice] scan 0c: ;Wolf; ';0204;8080
2024-04-17 12:45:01.158 [update notice] received unknown BC cmd: 71fe50180e0080008000800080008000800000
2024-04-17 12:45:01.183 [main error] HTTP failure: receive error: 404 Not Found
2024-04-17 12:45:01.183 [main error] unable to load scan config 0c: list files in wolf ERR: element not found
2024-04-17 12:45:01.183 [main error] scan config 0c: ERR: element not found
2024-04-17 12:45:03.460 [bus notice] scan 76: ;Kromschroeder;  ;0227;-
2024-04-17 12:45:03.460 [update notice] store 76 ident: done
2024-04-17 12:45:03.460 [update notice] sent scan-read scan.76  QQ=31: Kromschroeder;  ;0227;-
2024-04-17 12:45:03.460 [bus notice] scan 76: ;Kromschroeder;  ;0227;-
2024-04-17 12:45:03.671 [main error] HTTP failure: receive error: 404 Not Found
2024-04-17 12:45:03.671 [main error] unable to load scan config 76: list files in kromschroeder ERR: element not found
2024-04-17 12:45:03.671 [main error] scan config 76: ERR: element not found
2024-04-17 12:45:16.036 [update notice] received unknown BC cmd: 71fe50171041d91304da0200800080008000800080
2024-04-17 12:45:20.090 [update notice] received unknown BC cmd: 71fe50230998620901005d010000
2024-04-17 12:45:30.723 [update notice] received unknown BC cmd: 71fe50180e0080008000800080008000800000
2024-04-17 12:45:38.768 [bus notice] new master 0f, master count 4
2024-04-17 12:45:38.768 [update notice] received update-read broadcast datetime QQ=0f: -;12:45:10;17.04.2024
2024-04-17 12:45:45.826 [main error] scan config 14: ERR: read timeout
2024-04-17 12:45:46.111 [update notice] received unknown BC cmd: 71fe50171041d91304db0200800080008000800080
2024-04-17 12:45:50.033 [update notice] received unknown BC cmd: 71fe50230998620901005d010000
2024-04-17 12:46:01.048 [update notice] received unknown BC cmd: 71fe50180e0080008000800080008000800000
2024-04-17 12:46:04.146 [update notice] received unknown MS cmd: 0776502203cc0100 / 020000
2024-04-17 12:46:04.385 [main error] scan config 14: ERR: read timeout
2024-04-17 12:46:04.526 [update notice] received unknown MS cmd: 0776502203ccc103 / 020100
2024-04-17 12:46:04.881 [update notice] received unknown MS cmd: 0776502203cc4801 / 021500
2024-04-17 12:46:05.236 [update notice] received unknown MS cmd: 0776502203cc6209 / 020100
2024-04-17 12:46:05.591 [update notice] received unknown MS cmd: 0776502203cc2d0a / 020080
2024-04-17 12:46:05.948 [update notice] received unknown MS cmd: 0776502203cc2b0a / 0270fe
2024-04-17 12:46:06.301 [update notice] received unknown MS cmd: 0776502203cc300a / 020080
2024-04-17 12:46:06.631 [main notice] scan completed 1 time(s), check again
2024-04-17 12:46:06.640 [update notice] received unknown MS cmd: 0776502203ccae09 / 020000
2024-04-17 12:46:06.994 [update notice] received unknown MS cmd: 0776502203ccf802 / 020000
2024-04-17 12:46:07.347 [update notice] received unknown MS cmd: 0776502203cc2f0a / 020080
2024-04-17 12:46:16.327 [update notice] received unknown BC cmd: 71fe50171041d91304db0200800080008000800080
2024-04-17 12:46:20.383 [update notice] received unknown BC cmd: 71fe50230998620901005d010000
2024-04-17 12:46:31.396 [update notice] received unknown BC cmd: 71fe50180e0080008000800080008000800000
2024-04-17 12:46:38.540 [update notice] received update-read broadcast datetime QQ=0f: -;12:46:10;17.04.2024

**Comment by asmaps** [comment]
#950 helped a lot. At least I have some basic fields working now, working on getting more to work.

**Comment by valonsodev** [comment]
Hi there, i'll probably tackle what @zivillian talked about in the near future about generating the config files from the xml files. 

So it would probably go like:
XML -> JSON -> TypeSpec -> csv

Few things:

- I have no idea how the ism7 device identifies WOLF tech. Doing some reverse engineering of the app dll's brings up some weird stuff using known bus addresses and retrieving some type of special values? If anyone wants to see its inside `Wolf.Common.BusconfigInterfaceBusconfigMetaData`.
This means that decyphering the address for the `@zz()` of the TypeSpec files will be a bit of a problem.

- My boiler is freaking old (~20 years), reports as kromschroeder and for some reason EBUS messages just dont appear on my bus. I have some old wolf devices that act as Remote controls and their EBUS messages are not visible anywhere. I can only see what i believe to be some synchronization messages between the two remote controls.

I havent used ebusd a lot but from what i can see around forums and in ebusd wolf has two "special" messages:
- `@base(0x50, 0x22)` for reads
- `@base(0x50, 0x23)` for writes
I dont know if anything else is needed for other weird parameters¿ But again i have no way to test this.

I now provide a more reasonable way to inspect the xml files, a json file compiled from the xml files in case anyone wants to take a look at them if they are missing definitions.
[wolf.json](https://github.com/user-attachments/files/21865875/wolf.json)


LIST OF WOLF DEVICES THE APP RECOGNIZES:
Decypher what you can of the names, for example COB is Oil Condensing Boiler
```
// DeviceTemplateId - Name
10000 - DHK_BM
20000 - MK_BM
30000 - BM
40000 - MM
50000 - KM
60000 - SM1
70000 - SM2
80000 - CGB
90000 - COB
100000 - CGG-2
110000 - CGU-2
120000 - R1-2
130000 - R3
140000 - R21
150000 - BWS-1
160000 - BWL-1
170000 - BWW-1
180000 - CGB-2
190000 - ISM
200000 - MGK-2
210000 - TOB
220000 - BM-2
230000 - CWL
240000 - BWL-1S
250000 - FGB
260000 - ZirkSol_BM-2
270000 - CHA
280000 - COB-2
290000 - CGG-3
340000 - DHK_BM-2
350000 - WWSystem_BM-2
360000 - MK_BM-2
370000 - WW_BM-2
380000 - CGB-2-3855
390000 - CWL-2
400000 - RM-2
410000 - DHK_RM-2
420000 - WWSystem_RM-2
430000 - CGB-2-75100
440000 - TGB-2
450000 - DFB
460000 - FHA
660000 - WRS-K
670000 - WRS-HKVS
680000 - WRS-IK
690000 - WRS-K Pool
700000 - WPS-BHKW
710000 - WRS-CKL Pool
```

---

My device is a COB this means that i can search for that in the json:
```json
{
    "DTID": "90000",
    "name": "COB",
    "parameters": [
```
Then inside the parameters array you can find the parameter you want to read/write by their name, for example "Boiler water temperature":
```json
{
    "type": "NumericParameterDescriptor",
    "PTID": "90001",
    "name": {
        "DEU": "Kesseltemperatur",
        "ESP": "Temperatura caldera",
        "GBR": "Boiler water temperature"
    },
    "RemoveConditionId": "False",
    "ReadOnlyConditionId": "True",
    "InactiveConditionId": "False",
    "IsSnapshotTransmitEnabled": "false",
    "Decimals": "1",
    "ControlType": "NumericInput",
    "converter": {
        "type": "NumericConverter16Template",
        "TelegramNr": "13",
        "Type": "SS10"
    },
    "read_command": "CC0D00"
}
```
Again i have no clue what half of the things mean and its even missing many others that i didnt believe to be relevant, but they seem pretty self explanatory. ReadOnlyConditionId probably means that the parameter is read only.

Then "converter" is the important part, it tells you how to read/write the value, in this case our dear neighbor zivillian already did the hard part in https://github.com/zivillian/ism7mqtt

So we know that TelegramNr is what you need to send transformed the following way (Python for convenience):
```python
def telegram_to_read_command(telegram_num):
    # Convert to hex, little endian, prefix with CC
    hex_str = f"{int(telegram_num):04x}"
    little_endian = hex_str[2:] + hex_str[:2]
    return "CC" + little_endian.upper()
```
So 17 -> Gives us 0x0D00 -> CC0D00
(It is already in the json for convenience)

According to https://github.com/john30/ebusd/issues/167 the CC part is irrelevant since its a CRC that wolf devices ignore so the TypeSpec template defaults it to 00

Then when you get the ebus response you need to convert it based on the "Type", again zivillian did the hard part, the options are:
```csharp
case "US":
    result = JsonValue.Create(_value.Value);
    break;
case "SS":
    result = JsonValue.Create((short)_value.Value);
    break;
case "SS10":
    result = JsonValue.Create(((short)_value.Value) / 10.0);
    break;
case "US10":
    result = JsonValue.Create(_value.Value / 10.0);
    break;
case "SS100":
    result = JsonValue.Create(((short)_value.Value) / 100.0);
    break;
case "SSPR":
    result = JsonValue.Create(((short) _value.Value) * (1.0 / 255));
    break;
case "US4":
    result = JsonValue.Create(_value.Value / 4.0);
    break;
case "IntDiv60":
    result = JsonValue.Create(_value.Value / 60.0);
    break;
```
I havent looked what the equivalent is in ebusd for all of them but SS10 is temp10 in the TypeSpec.

Full TypeSpec example that worked for my boiler:
```typespec
import "@ebusd/ebus-typespec";
import "./_templates.tsp";
import "../opdata_inc.tsp";
using Ebus;
using Ebus.Num;
using Ebus.Dtm;
using Ebus.Str;

namespace Wolf;
// 0x35 is my boiler, pure guess tho, i know its this one because of the few messages i can see in the bus
@zz(0x35) // 
namespace hc {
  /** Boiler temperature */
  @id(0x0D, 0x00) // Message 0D00
  @poll(2)
  model BoilerTemp is ReadonlyRegister<temp10>;
  /** Accumulator temperature */
  @id(0x0E, 0x00) // Same reasoning as above but with TelegramNr 14 thus 0E00
  @poll(2)
  model AcumulatorTemp is ReadonlyRegister<temp10>;
}
```
This "compiled" to this csv:
`35..bc.csv`
```csv
type,circuit,level,name,comment,qq,zz,pbsb,id,*name,part,type,divisor/values,unit,comment
*r,
r2,Hc,,BoilerTemp,Boiler temperature,,,5022,000d00,value,,UIN,10,°C,temperature
r2,Hc,,AcumulatorTemp,Acumulator temperature,,,5022,000e00,value,,UIN,10,°C,temperature
```
I havent tried writing yet, but the json contains all possible values for write parameters:
```json
{
    "type": "ListParameterDescriptor",
    "PTID": "410010",
    "name": {
        "DEU": "Programmwahl",
        "ESP": "Selección de programa",
        "GBR": "Prog. select."
    },
    "RemoveConditionId": "False",
    "ReadOnlyConditionId": "False",
    "InactiveConditionId": "False",
    "IsSnapshotTransmitEnabled": "true",
    "MinValueCondition": "0",
    "MaxValueCondition": "3",
    "ControlType": "ProgramSelectionListView",
    "KeyValueList": [
        {
        "key": "0",
        "value": {
            "DEU": "Standby",
            "ESP": "Standby",
            "GBR": "Standby"
        }
        },
        {
        "key": "1",
        "value": {
            "DEU": "Auto",
            "ESP": "Auto",
            "GBR": "Auto"
        }
        },
        {
        "key": "2",
        "value": {
            "DEU": "Sommer",
            "ESP": "Verano",
            "GBR": "Summer"
        }
        }
    ],
    "converter": {

        "type": "NumericConverter16Template",
        "TelegramNr": "10100",
        "Type": "US"
    },
    "read_command": "CC7427"
}
```

If anyone knows how ISM7 identifies the devices it would be of great help. If its not possible i will just probably create the TypeSpec without `@zz` for people to fill manually or something.

**Reply by audaciousengineer** [reply]
Thank you @valonsodev for your detailed steps and for sharing this great discovery by @zivillian! Your description helped me a lot.
Thanks to your json file, I was able to decrypt many (but not all) messages seen on the bus. Interestingly, some message patterns belong to devices not present in my setup. However, looking into this, I found some telegrams to be shared across devices.

The 5800 entries in the json file comprise 633 different commands, most of which have a unique meaning, even across different devices. Unfortunately, about 193 are duplicates whose meaning depends on the receiving device. Thus, it still remains to identify the devices at some point. 

I want to share this information as it could be helpful to build a common config for the WOLF devices. But I am not yet sure how to proceed from here.

**Comment by andrasg** [comment]
Hi everyone, great thread! I have a working Wolf CGW2 config, able to read quite a lot of data using the following csv:

[08.csv](https://github.com/user-attachments/files/23528526/08.csv)

I only see 1 broadcast message on the bus, but can read values using polling and also via the force read in the command line.

I have a wiring question though. I have my AM module integrated into the boiler (snapped in place into the unit) and have my ebus adapter wired to the ebus connection on the PCB. Is this the right way to do it? Or should my ebus adapter be wired "between" the boiler and the AM module?

I was not able to write anything to the boiler (wanted to set DHW target), I get an error:

```
2025-11-13 16:12:18.508 [bus error] send to 03: ERR: read timeout, retry
```

Anyone got writing of settings working?

**Reply by audaciousengineer** [reply]
I had success when I wrote to the BM2 instead of the boiler directly. So, instead of 03, try to write to 35 (BM2 address).
In my opinion, some write commands are there only to indicate that a write command was seen on the bus. But for some reason, you cannot use them directly. 

I 'quick and dirty' added this line to my 35..bm2.csv and it works:
`w | bm2 | wwsoll | Warmwasser Solltemperatur |   | 35 | 5023 | 001300 | value |   | UIN | 10`

**Reply by andrasg** [reply]
I don't have a BM2. My AM is snapped into the boiler itself and my ebusd is connected to the ebus terminals in the boiler. I am not seeing the AM on the bus unfortunately. I was not able to figure out how to send a warmwater settemp command to the boiler. Heck, I was not able to send any write commands at all.

**Reply by Ces1254** [reply]
Did you check whether your writing command actually works, despite the read timeout error? I had the same issue writing on a Wolf system (an FHA, actually, not a CGW2) and discovered that there was actually no answer to the master-slave command, despite the writing command was actually correctly executed. See the issue I raised [here](https://github.com/john30/ebusd/issues/1696). Best

**Reply by realResi** [reply]
For CGG-2 with BM: writing HG parameters to ZZ=03 (Feuerungsautomat) 
works with PBSB=5023 and suffix 9d010000 — no read timeout in our case.

Ces1254 is right though: the read timeout on write doesn't necessarily 
mean the command failed. Worth checking the actual value after the command.

For andrasg without BM2: the AM module snapped into the boiler is likely 
on the boiler's internal bus rather than the external eBUS — that would 
explain why it's invisible to ebusd.

More details on what works/doesn't work for CGG-2 in our config:
https://github.com/realResi/ebusd-wolf-cgg2

**Comment by roburb** [comment]
Hi all,
I would like to ask for help.
Can you share with me csv file I can use with CHA 07 and BM-2 please?
I have read the process of creating csv file "XML -> JSON -> TypeSpec -> csv" but it looks to hard for me.
Thank you

**Reply by Ces1254** [reply]
Hi there!
If it may help, you can check [here](https://github.com/Ces1254/Wolf_ebusd_config). These are the config files I use with my Wolf FHA. Being newer than the CHA, I guess that most of the message definitions are valid also for the CHA. I would also been interested to know the experience from other users in writing to the BM-2 controller. It works for me, but ebusd reports an error. See the issue I raised [here](https://github.com/john30/ebusd/issues/1696).
Best

**Comment by martinlbb** [comment]
Hi guys,

I answer here, as I plugged yesterday my **Solar Module SM2 (with BM2 screen)** via eBus (device **EBUS to WiFi Adapter Module V6.3**) to my home assistant server running MQTT and ebusd.

At first, I struggle to get it recognized, and finally I have the values I am looking for: (**collector temp, solar tank exchanger temp, solar return line temp, pump state, and even pump hour meter**).
![mqtt-ebusd-wolf-sm2-bm2-sensors](https://github.com/user-attachments/assets/adbb7d0c-b3ec-4f18-8f22-ff4e37bac9d0)

### My install SM2/BM2 old gen
Although this is a new installation, I installed an old SM2 gen1 device I recovered. Firmware is old (228-02 on old SM2 motherboard).
It is close to SM1 device by spirit and limited functionality.
I have 10 square meter of thermal solar receiver, an external water/water heat exchanger, two cycling pumps and a 1000L tank (heating water and a heat exchanger immersed within the tank for on-demand domestic hot water).

First of all, when configuring ebusd, I had to ignore enhanced mode (can't connect to bus), and switch to regular mode. In ebusd configuration, line "Network adapter address', I get rid of enh (enh:IP:port) and use IP:port instead (ie: 192.168.1.34:3333) with specific 3333 port (regular mode, read/write).

At this stage, device was talking to ebusd and gives me warning about **Kromschroeder** device ID 76 not recognized. I found quickly that it is the same brand as Wolf, and a direct symbolic link exist in CSV conf.

### Working CSV File
**After some struggling, I wrote this CSV file.**
It is okay for a single tank, single collector installation.
[76..sm.csv](https://github.com/user-attachments/files/26091099/76.sm.csv)

Upload that file to your home assistant server in `/config/ebusd-configuration/latest/en/kromschroeder`.
Then go to ebusd _advanced_ configuration, and set config_path to `/config/ebusd-configuration/latest/en`
It should run flawlessly after restarting app.
If you are already using CSV definitions from ebsud, just drop this file in wolf directory.

### Values does not appears in MQTT device (solved)
I was annoyed with values not showing in MQTT device, but available in MQTT data (using MQTT explorer).
I realized that auto-discovery use a filtering file (`/etc/ebusd/mqtt-hassio.cfg`).
If some value has a name not accepted by a keyword list (`status|temp|humidity|yield|count|energy|power|runtime|hours|starts|mode|curve|^load$|^party$|sensor|timer`), it will never be shown!
I have updated name value to have it nicely shown in MQTT device (beware! French HA screenshot):
![mqtt-ebusd-wolf-sm2-bm2](https://github.com/user-attachments/assets/eab160d3-b0d4-42c8-8be4-1e323a66309b)

### Other values
I tried map other values, like **yield day, yield total**, but it does not work (returning 0).
I can map some settings (**max collector temp, max tank temp, heat yield config, flow rate**, ...) but it was useless for me. I need to map and follow real world values, not never ever changing settings.
On newer device (SM gen2 with newer firmware), you should be able to map Yield values.
 

Hope this helps!

**Comment by realResi** [comment]
I've been running a Wolf CGG-2-18 with BM control module on ebusd for a while now and finally got around to publishing the configuration.

Full stack including MQTT auto-discovery for Home Assistant:
https://github.com/realResi/ebusd-wolf-cgg2

A lean version (CSV only) was also merged into puni2k's collection:
https://github.com/puni2k/ebusd-configuration-wolf/pull/12

Happy to answer questions.

**Comment by realResi** [comment]
One more finding: the Wolf BM has no battery-backed RTC and loses its time and date 
after any power cut. With ebusd you can correct it automatically by spoofing 
QQ=0f (Funk-Empfänger address) via hex -s on TCP port 8888 — the BM only 
accepts datetime broadcasts from QQ=0f or QQ=30.

Documented with shell command and HA automation notes here:
https://github.com/realResi/ebusd-wolf-cgg2

**Reply by katagia** [reply]
The BM has no battery but is has a goldcap. Mine was broken, afer replacing it, it keeps its time.

**Reply by realResi** [reply]
Good to know — thanks for the hardware tip! The goldcap explains a lot.

The software workaround via ebusd is still useful as a fallback for extended outages where even a healthy goldcap runs out, or simply for those who prefer not to open up the boilers "brain" physically. ;)

**Comment by homa57** [comment]
Some findings from a **Wolf CGU-2-10 + old BM** (ebusd 26.1 HA add-on, adapter v5), in case they help others:

**1. Output A1 (circulation pump) is readable at the boiler**
TelegramNr 371 at slave `08`, **bit 6 = output A1** (HG14=3, circulation 20 %). Verified against the "A1" indicator on the BM over several on/off cycles (`40 00` = on, `00 00` = off).
```
r1,kessel,status_relais,Relais (Tel. 371),,08,5022,117301,bit0,,BI0,,,,bit1,,BI1,,,,bit2,,BI2,,,,bit3,,BI3,,,,bit4,,BI4,,,,bit5,,BI5,,,,zirkulation,,BI6,0=aus;1=an,,,bit7,,BI7,,,,high,,HEX:1,,,
```

**2. BM DHW setpoint and operating mode – read AND write at slave `35`, no suffix needed**
```
r,bm,wwsolltemp,DHW setpoint,,35,5022,001300,value,,SIN,10,°C,
w,bm,wwsolltemp,DHW setpoint,,35,5023,001300,value,,SIN,10,°C,
r,bm,betriebsart,Operating mode,,35,5022,091201,value,,UIN,1=aus;2=zeituhr;3=heizbetrieb;4=sparbetrieb;5=nur_warmwasser,,
w,bm,betriebsart,Operating mode,,35,5023,001201,value,,UIN,1=aus;2=zeituhr;3=heizbetrieb;4=sparbetrieb;5=nur_warmwasser,,
```
Mode values verified at the BM: 1 = off, 2 = timer, 3 = sun+DHW, 4 = moon+DHW, 5 = DHW only. Changes show up on the BM display immediately.

**3. Heating vs. DHW demand from broadcast `f1 fe 0503`**
Second data byte: **bit 0 = heating**, **bit 2 = DHW** (verified: DHW only / heating only / both / DHW finished). Combined with the flame bit from `03 fe 0503` this splits gas consumption cleanly into heating / DHW / cooking.

Thanks to @dktzde for the CRC8 finding and to @realResi – both saved a lot of guessing.

_homa
